static/app/components/replays/canvasReplayerPlugin.tsx

```diff
@@ -67,16 +160,34 @@ export function CanvasReplayerPlugin(events: eventWithTime[]) {
     return cloneNode;
   }
 
-  const promises: Promise<any>[] = [];
-  for (const event of events) {
-    if (
-      event.type === EventType.IncrementalSnapshot &&
-      event.data.source === IncrementalSource.CanvasMutation
-    ) {
-      promises.push(deserializeAndPreloadCanvasEvents(event.data, event));
```

Comment:
This was one of the crash causes, where we would need at least 2 promises per event. If you changed this to be serial so that you wait for `deserializeAndPreloadCanvasEvents` to finish before moving to the next event, then it would eventually crash due to too many images being stored in `canvasEventMap`.